From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 24 Feb 2022 04:22:32 +0100
Subject: sessionMode: Add support for configuring an icons resource name

Similar to what we did (and upstreamed) for the theme, in this case
however there's very likely no interest from upstream for this.

Forwarded: not-needed
---
 js/ui/main.js        | 2 +-
 js/ui/sessionMode.js | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/js/ui/main.js b/js/ui/main.js
index f9d4e9c..adab13c 100644
--- a/js/ui/main.js
+++ b/js/ui/main.js
@@ -615,7 +615,7 @@ export function reloadThemeResource() {
 
 /** @private */
 function _loadIcons() {
-    _iconResource = Gio.Resource.load(`${global.datadir}/gnome-shell-icons.gresource`);
+    _iconResource = Gio.Resource.load(`${global.datadir}/${sessionMode.iconsResourceName}`);
     _iconResource._register();
 }
 
diff --git a/js/ui/sessionMode.js b/js/ui/sessionMode.js
index 3e4f8b8..d884945 100644
--- a/js/ui/sessionMode.js
+++ b/js/ui/sessionMode.js
@@ -24,6 +24,7 @@ const _modes = {
         stylesheetName: 'gnome-shell.css',
         colorScheme: 'prefer-dark',
         themeResourceName: 'gnome-shell-theme.gresource',
+        iconsResourceName: 'gnome-shell-icons.gresource',
         hasOverview: false,
         showCalendarEvents: false,
         showWelcomeDialog: false,
